### DISCUSSION TITLE
Use a single instance for improve performance.

### DISCUSSION THREAD

**Discussion by byehack** [encode/httpx#2559] [Q&A] [answered]
```py
import asyncio, httpx


async def req(url):
    request = httpx.Request("GET", url)
    resp = await client.send(request)
    request = httpx.Request("GET", url, cookies=resp.cookies)
    resp = await client.send(request)


client = httpx.AsyncClient()
urls = [...]
asyncio.run(asyncio.gather(*[req(url) for url in urls]))
```

Are we allowed to use single main instance and use it with different request options?

**Comment by zanieb** [comment, marked as answer]
Hey @byehack — yes that is recommended but you'll want to context manage the client


```python
async def main:
    async with httpx.AsyncClient() as client:
        ....
```

See https://www.python-httpx.org/async/#making-async-requests

**Comment by byehack** [comment]
Thanks @madkinsz 
True, I read documentation.
but in presentation I noticed it uses high memory and CPU usages. (when huge sending requests)
it didn't clarify in document that `.send` method will affect `client` instance or not. (like [persistance](https://www.python-httpx.org/advanced/#extra-features))
before do changes in my code, I wanted to ensure that it doesn't make problems.
